### PR TITLE
Replace Lidgren source-copy with submodule

### PR DESCRIPTION
**Pull request by Therzok**
This is a WIP patch which moves to using upstream lidgren submodule.

First, we need [this patch](https://github.com/lidgren/lidgren-network-gen3/pull/36) merged into upstream.

I also need a review of the changes to the definition file. (The rest of the changes are just deletions/git metadata).

### CODE REVIEW

**Comment by Therzok** [comment]
The upstream patch is merged, awaiting CI to give its :+1: then I think it's good to go.

File: Build/Projects/Lidgren.Network.definition [hunk 1]
```diff
@@ -30,8 +30,8 @@
     <WarningLevel>0</WarningLevel>
     <MonoDevelopPoliciesFile>Build/MonoDevelopPolicies.xml</MonoDevelopPoliciesFile>
     <CustomDefinitions>
-    <!-- 
-    Regardless of the crappy name of the define... Lidgren uses UNITY_4_5 to signify 
+    <!--
```
Comment (Therzok) [inline Build/Projects/Lidgren.Network.definition]: Sorry for the whitespace changes, vim cleaned the file up on save.

**Comment by Therzok** [comment]
Also, please note, that I haven't backported any of the possible fixes that may have happened in monogame, but not have been merged upstream.

**Comment by dellis1972** [comment]
@Therzok I know for a fact that there are Android Changes to Lidgren that have NOT been merged upstream. Something to do with the NetworkManager I think... its been over two years since that code was changed.

**Comment by Therzok** [comment]
I'm going through the changes that happened to lidgren locally, since it got added:
 * csproj/sln changes, but those are superseded by protobuild.
 * Android logging - it's in upstream.
 * Missed [this](https://github.com/mono/MonoGame/commit/d1d4ba1a98d4843a51d707254726fdca5709081e#diff-84212be9cd4da3f33688d340c95f8edaR378) transformation to UTF8.
 * There was [PlatformSocket](https://github.com/mono/MonoGame/commit/68d09a2d796a69faf25c60c244e784f8ece53093) and PSM support, but they're gone as of now.

Sorry, but I really can't find any change. Went through all the commits that happened in the lidgren source copy.

**Comment by Therzok** [comment]
Submitted https://github.com/lidgren/lidgren-network-gen3/pull/41 to eliminate the last discrepancy.

**Pull request closed without merging**